### PR TITLE
Expose org.eclipse.jetty:jetty-servlet in bom.

### PR DESCRIPTION
**Pull request by wetted**

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -48,7 +49,7 @@ kotlin-reflect = { module = 'org.jetbrains.kotlin:kotlin-reflect' }
 
 tomcat-embed-core = { module = 'org.apache.tomcat.embed:tomcat-embed-core', version.ref = 'tomcat' }
 undertow-servlet = { module = 'io.undertow:undertow-servlet', version.ref = 'undertow' }
-jetty = { module = 'org.eclipse.jetty:jetty-servlet', version.ref = 'jetty' }
+managed-jetty = { module = 'org.eclipse.jetty:jetty-servlet', version.ref = 'managed-jetty' }
```
Comment (timyates) [inline gradle/libs.versions.toml:52]: Do we need to make this managed?

In the other linked PRs to this one, it's `jetty-server` not `jetty-servlet`

So could we just expose the version?

I don't know

**Review by timyates — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (sdelamo) [reply gradle/libs.versions.toml:52]: I think @timyates is right. @wetted change this PR to expose only the version.

**Review by sdelamo — CHANGES_REQUESTED** [review]
I think @timyates is right. @wetted change this PR to expose only the version.

**Comment by timyates** [comment]
Actually...thinking about it, I'm wrong...

There's no way in aws to hava a catalog entry for jetty-server with the version from here...

**Comment by wetted** [comment]
> Actually...thinking about it, I'm wrong...
> 
> There's no way in aws to hava a catalog entry for jetty-server with the version from here...

@sdelamo @timyates I think there's enough back and forth here that I don't understand what we should be doing.

**Comment by timyates** [comment]
@sdelamo @wetted How's this https://github.com/micronaut-projects/micronaut-servlet/pull/651/commits/44d2cf5267c01f78108a0dc55943992ab8c4f996

So here we expose a version and a BOM for jetty...

We can then (in AWS for example) change the libs.versions.toml thusly:

```diff
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index cdc245337..0ce4aa60c 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -7,7 +7,6 @@ spock = "2.3-groovy-4.0"

 bouncycastle = '1.70'
 fileupload = '0.0.6'
-jetty = '11.0.20'
 logback-json-classic = '0.1.5'

 micronaut-discovery = "4.2.0"
@@ -88,7 +87,7 @@ bouncycastle-provider = { module = 'org.bouncycastle:bcprov-jdk15on', version.re
 fileupload = { module = 'org.javadelight:delight-fileupload', version.ref = 'fileupload' }
 graal-sdk = { module = 'org.graalvm.sdk:graal-sdk', version.ref = 'graal' }
 jackson-afterburner = { module = 'com.fasterxml.jackson.module:jackson-module-afterburner' }
-jetty-server = { module = 'org.eclipse.jetty:jetty-server', version.ref = 'jetty' }
+jetty-server = { module = 'org.eclipse.jetty:jetty-server' }
 jcl-over-slf4j = { module = 'org.slf4j:jcl-over-slf4j', version.ref = 'slf4j' }
 junit-jupiter-engine = { module = 'org.junit.jupiter:junit-jupiter-engine' }
 junit-jupiter-api = { module = 'org.junit.jupiter:junit-jupiter-api' }
```

(so we don't declare a Jetty version in there), and then we would need to update the build files with platform dependencies

```diff
diff --git a/function-aws-api-proxy-test/build.gradle.kts b/function-aws-api-proxy-test/build.gradle.kts
index 05ef65887..1432c7bef 100644
--- a/function-aws-api-proxy-test/build.gradle.kts
+++ b/function-aws-api-proxy-test/build.gradle.kts
@@ -5,6 +5,7 @@ plugins {
 dependencies {
     api(mn.micronaut.http.server)
     api(projects.micronautFunctionAwsApiProxy)
+    implementation(platform(mnServlet.boms.jetty))
     implementation(libs.jetty.server)
     testImplementation(mn.micronaut.http.client)
     testImplementation(mn.micronaut.jackson.databind)
```

I *think* this is where we wanted to end up?

**Comment by wetted** [comment]
> @sdelamo @wetted How's this [44d2cf5](https://github.com/micronaut-projects/micronaut-servlet/commit/44d2cf5267c01f78108a0dc55943992ab8c4f996)
> 
> So here we expose a version and a BOM for jetty...
> 
> We can then (in AWS for example) change the libs.versions.toml thusly:
> ...
> 
> I _think_ this is where we wanted to end up?

Yeah, I think that looks good. Thanks.

**Pull request merged**